btree/btree.c

```diff
@@ -2973,6 +3004,28 @@ static void btree_node_update_credit(const struct m0_btree  *tree,
 	m0_be_tx_credit_add(accum, &M0_BE_TX_CREDIT(1, node_size));
 }
 
+
+/**
+ * This function will calculate credits required for the delete KV operation and
+ * add those credits to @accum.
+ */
+static void btree_del_credit(const struct m0_btree  *tree,
+			     struct m0_be_tx_credit *accum)
+{
+	struct m0_be_tx_credit cred = {};
+
+	/* Credits for freeing the node. */
+	node_free_credits(tree->t_desc->t_root, &cred);
+	m0_be_tx_credit_mac(accum, &cred, MAX_TREE_HEIGHT);
+
+	/**
+	 * Credits for update node:
+	 * After deletetion of record, if node is not underflowed, node will get
+	 * updated.
+	 */
+	btree_node_update_credit(tree, accum);
```

Comment:
This is not needed because if we did not underflow then the credits calculated above for the node free are more than enough to capture the node update if a record was deleted.

Reply:
Right. I have made this change.